alphacsc/update_d_multi.py

```diff
@@ -22,6 +19,30 @@ def squeeze_all_except_one(X, axis=0):
     return X.squeeze(axis=squeeze_axis)
 
 
+def check_solver_and_constraints(rank1, solver_d, uv_constraint):
+
+    if rank1:
+        if solver_d == 'auto':
+            solver_d = 'alternate_adaptive'
+        if 'alternate' in solver_d:
+            if uv_constraint == 'auto':
+                uv_constraint = 'separate'
+            else:
+                assert uv_constraint == 'separate', (
+                    "solver_d='alternate*' should be used with "
+                    f"uv_constraint='separate'. Got '{uv_constraint}'."
+                )
+        elif uv_constraint == 'auto' and solver_d in ['joint', 'fista']:
+            uv_constraint = 'joint'
+    else:
+        assert solver_d in ['auto', 'fista'] and uv_constraint == 'auto', (
+            "If rank1 is False, uv_constraint should be 'auto' "
+            f"and solver_d should be auto or fista. Got {uv_constraint}"
```

Comment:
maybe output the value of `solver_d` here as well?
```suggestion
            f"and solver_d should be auto or fista. Got {solver_d=}, {uv_constraint=}"
```